src/main/java/com/synopsys/integration/alert/channel/email/EmailAddressHandler.java

```diff
@@ -0,0 +1,122 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.channel.email;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.channel.email.descriptor.EmailDescriptor;
+import com.synopsys.integration.alert.common.configuration.FieldAccessor;
+import com.synopsys.integration.alert.common.model.AggregateMessageContent;
+import com.synopsys.integration.alert.database.api.configuration.model.ConfigurationFieldModel;
+import com.synopsys.integration.alert.database.provider.blackduck.data.BlackDuckProjectEntity;
+import com.synopsys.integration.alert.database.provider.blackduck.data.BlackDuckProjectRepositoryAccessor;
+import com.synopsys.integration.alert.database.provider.blackduck.data.BlackDuckUserEntity;
+import com.synopsys.integration.alert.database.provider.blackduck.data.BlackDuckUserRepositoryAccessor;
+import com.synopsys.integration.alert.database.provider.blackduck.data.relation.UserProjectRelation;
+import com.synopsys.integration.alert.database.provider.blackduck.data.relation.UserProjectRelationRepositoryAccessor;
+import com.synopsys.integration.alert.provider.blackduck.BlackDuckProvider;
+
+@Component
+public class EmailAddressHandler {
+    private static final Logger logger = LoggerFactory.getLogger(EmailAddressHandler.class);
+
+    private final BlackDuckProjectRepositoryAccessor blackDuckProjectRepositoryAccessor;
+    private final UserProjectRelationRepositoryAccessor userProjectRelationRepositoryAccessor;
+    private final BlackDuckUserRepositoryAccessor blackDuckUserRepositoryAccessor;
+
+    @Autowired
+    public EmailAddressHandler(final BlackDuckProjectRepositoryAccessor blackDuckProjectRepositoryAccessor,
+        final UserProjectRelationRepositoryAccessor userProjectRelationRepositoryAccessor, final BlackDuckUserRepositoryAccessor blackDuckUserRepositoryAccessor) {
+        this.blackDuckProjectRepositoryAccessor = blackDuckProjectRepositoryAccessor;
+        this.userProjectRelationRepositoryAccessor = userProjectRelationRepositoryAccessor;
+        this.blackDuckUserRepositoryAccessor = blackDuckUserRepositoryAccessor;
+    }
+
+    public FieldAccessor updateEmailAddresses(final String provider, final AggregateMessageContent content, final FieldAccessor originalAccessor) {
+        if (StringUtils.isBlank(provider)) {
+            return originalAccessor;
+        }
+        if (BlackDuckProvider.COMPONENT_NAME.equals(provider)) {
```

Comment:
Might be easier to comprehend in the future if we separate provider specific content out into different classes or at least separate methods. Or better yet, we can add the option for email functionality to providers then get the provider and call their own functionality.